_test/tests/inc/common_saveWikiText.test.php

```diff
@@ -175,250 +292,238 @@ function test_savesequencedeleteexternalrevision() {
-        saveWikiText($page, 'teststring2long', 'third save', false);
+        // 3.2 external edit
+        file_put_contents($file, 'teststring external edit');
         clearstatcache(false, $file);
         $newmod = filemtime($file);
         $this->assertNotEquals($lastmod, $newmod);
         $lastmod = $newmod;
-        $revertrev = $newmod;
+        $expectExternal = array(
+            'date' => $lastmod,
+            'type' => DOKU_CHANGE_TYPE_CREATE,
+            'sum'  => 'created - external edit',
+            'sizechange' => 24,
+        );
 
         $pagelog = new PageChangeLog($page);
         $revisions = $pagelog->getRevisions(-1, 200);
-        $this->assertEquals(2, count($revisions));
-        $revinfo = $pagelog->getRevisionInfo($revisions[0]);
-        $this->assertEquals('third save', $revinfo['sum']);
-        $this->assertEquals(DOKU_CHANGE_TYPE_EDIT, $revinfo['type']);
-        $this->assertEquals(5, $revinfo['sizechange']);
+        $this->assertCount(0, $revisions); // external edit is not yet in changelog
+        // last revision
+        $this->assertFalse($pagelog->lastRevision(), 'changelog file does not yet exist');
+        // current revision
+        $currentRevInfo = $pagelog->getCurrentRevisionInfo();
+        $this->assertArrayHasKey('timestamp', $currentRevInfo, 'should be external revision');
+        $expectExternal += $currentRevInfo;
+        $this->assertEquals($expectExternal, $currentRevInfo);
+        // attic
+        $this->assertFileNotExists(wikiFN($page, $currentRevInfo['date']),'page does not yet exist in attic');
 
-        $this->waitForTick(); // wait for new revision ID
+        $this->waitForTick(true); // wait for new revision ID
 
-        // add a minor edit (unauthenticated)
-        saveWikiText($page, 'teststring3long', 'fourth save', true);
+        // 3.3 save on top of external edit
+        saveWikiText($page, 'teststring1', 'Test 3, first save', false);
         clearstatcache(false, $file);
         $newmod = filemtime($file);
         $this->assertNotEquals($lastmod, $newmod);
         $lastmod = $newmod;
+        $expect = array(
+            'date' => $lastmod,
+            'type' => DOKU_CHANGE_TYPE_EDIT,
+            'sum'  => 'Test 3, first save',
+            'sizechange' => -13,
+        );
 
         $pagelog = new PageChangeLog($page);
         $revisions = $pagelog->getRevisions(-1, 200);
-        $this->assertEquals(3, count($revisions));
-        $revinfo = $pagelog->getRevisionInfo($revisions[0]);
-        $this->assertEquals('fourth save', $revinfo['sum']);
-        $this->assertEquals(DOKU_CHANGE_TYPE_EDIT, $revinfo['type']);
-        $this->assertEquals(0, $revinfo['sizechange']);
+        $this->assertCount(2, $revisions); // two more revisions now!
+        // last revision
+        $lastRevInfo = $pagelog->getRevisionInfo($revisions[0]);
+        $expect += $lastRevInfo;
+        $this->assertEquals($expect, $lastRevInfo);
```

Comment:
I expect that the changelog has now stored a external create, but in the past this was different. Therefore a test would be useful. 
e.g.
```
$beforeRevInfo = $pagelog->getRevisionInfo($revisions[1]);
$expectExternal += $beforeRevInfo; //union works probably not as expected, because it is the second union.
$this->assertEquals($expectExternal, $beforeRevInfo);
```

Reply:
test added see  e21f947